src/main/java/org/apache/mesos/scheduler/SchedulerDriverFactory.java

```diff
@@ -0,0 +1,131 @@
+   *     {@link org.apache.mesos.Protos.Credential}, ignored if {@code null}/empty
+   * @return A {@link SchedulerDriver} configured with the provided info
+   * @throws IllegalArgumentException if {@link FrameworkInfo}.principal is unset or empty when
+   *     authentication is needed
+   */
+  public SchedulerDriver create(
+      final Scheduler scheduler,
+      final FrameworkInfo frameworkInfo,
+      final String masterUrl,
+      final byte[] credentialSecret) {
+    Credential credential;
+    if (credentialSecret != null && credentialSecret.length > 0) {
+      // User has manually provided a Secret. Provide a Credential with Principal + Secret.
+      // (note: we intentionally avoid logging the content of the credential secret, just in case)
+      LOGGER.info("Creating secret authenticated MesosSchedulerDriver for "
+          + "scheduler[{}], frameworkInfo[{}], masterUrl[{}], credentialSecret[{} bytes]",
+          scheduler, frameworkInfo, masterUrl, credentialSecret.length);
+      credential = Credential.newBuilder()
+          .setPrincipal(getPrincipal(frameworkInfo, "secret"))
+          .setSecretBytes(ByteString.copyFrom(credentialSecret))
+          .build();
+    } else if (isSideChannelActive()) {
+      // Sidechannel auth is enabled. Provide a Credential with only the Principal set.
+      LOGGER.info("Creating sidechannel authenticated MesosSchedulerDriver for "
+          + "scheduler[{}], frameworkInfo[{}], masterUrl[{}]",
+          scheduler, frameworkInfo, masterUrl);
+      credential = Credential.newBuilder()
+          .setPrincipal(getPrincipal(frameworkInfo, "sidechannel"))
+          .build();
+    } else {
+      // No auth. Provide no credential.
+      LOGGER.info("Creating unauthenticated MesosSchedulerDriver for "
+          + "scheduler[{}], frameworkInfo[{}], masterUrl[{}]",
+          scheduler, frameworkInfo, masterUrl);
+      credential = null;
+    }
+    return createInternal(scheduler, frameworkInfo, masterUrl, credential);
+  }
+
+  /**
+   * Broken out into a separate function to allow testing with custom SchedulerDrivers.
+   */
+  protected SchedulerDriver createInternal(
+      final Scheduler scheduler,
+      final FrameworkInfo frameworkInfo,
+      final String masterUrl,
+      final Credential credential) {
+    if (credential == null) {
+      return new MesosSchedulerDriver(scheduler, frameworkInfo, masterUrl);
+    } else {
+      return new MesosSchedulerDriver(scheduler, frameworkInfo, masterUrl, credential);
+    }
+  }
+
+  /**
+   * Returns whether it appears that side channel auth should be used when creating the
+   * SchedulerDriver. Side channel auth may take the form of Bouncer-based framework authentication
+   * or potentially other methods in the future (eg Kerberos). Broken out into a separate function
+   * to allow easy customization in tests.
+   */
+  protected boolean isSideChannelActive() {
+    return System.getenv(SIDECHANNEL_AUTH_ENV_NAME) != null;
```

Comment:
And to add: due to how Universe works, each framework would also need to independently define that flag in their packaging, leaving a burden on the framework developer and leaving open a window for inconsistencies in how the flag is defined

Given both of the above, having the library deal with it automatically feels like the preferable option